src/Compilers/CSharp/Portable/Symbols/TypeSymbol.cs

```diff
@@ -980,19 +992,41 @@ private static Symbol ComputeImplementationForInterfaceMember(Symbol interfaceMe
 
             Debug.Assert(!canBeImplementedImplicitly || (object)implementingBaseOpt == null);
 
+            bool tryDefaultInterfaceImplementation = true;
+
             // Dev10 has some extra restrictions and extra wiggle room when finding implicit
             // implementations for interface accessors.  Perform some extra checks and possibly
             // update the result (i.e. implicitImpl).
             if (interfaceMember.IsAccessor())
             {
-                // https://github.com/dotnet/roslyn/issues/34453: Do we need to adjust behavior of this function in any way?
-                CheckForImplementationOfCorrespondingPropertyOrEvent((MethodSymbol)interfaceMember, implementingType, implementingTypeIsFromSomeCompilation, ref implicitImpl, ref useSiteDiagnostics);
+                Symbol originalImplicitImpl = implicitImpl;
+                CheckForImplementationOfCorrespondingPropertyOrEvent((MethodSymbol)interfaceMember, implementingType, implementingTypeIsFromSomeCompilation, ref implicitImpl);
+
+                // If we discarded the candidate, we don't want default interface implementation to take over later, since runtime might still use the discarded candidate.
+                if (originalImplicitImpl is object && implicitImpl is null)
+                {
+                    tryDefaultInterfaceImplementation = false;
+                }
             }
 
-            if ((object)implicitImpl == null && seenTypeDeclaringInterface)
+            Symbol defaultImpl = null;
+
+            if ((object)implicitImpl == null && seenTypeDeclaringInterface && tryDefaultInterfaceImplementation)
+            {
+                if (ignoreImplementationInInterfaces)
+                {
+                    implementationInInterfacesMightChangeResult = true;
+                }
+                else
+                {
+                    // Check for default interface implementations
+                    defaultImpl = FindMostSpecificImplementationInInterfaces(interfaceMember, implementingType, ref useSiteDiagnostics, diagnostics);
+                    implementationInInterfacesMightChangeResult = false;
+                }
+            }
+            else
             {
-                // Check for default interface implementations
-                implicitImpl = FindMostSpecificImplementationInInterfaces(interfaceMember, implementingType, ref useSiteDiagnostics, diagnostics);
+                implementationInInterfacesMightChangeResult = false;
             }
```

Comment:
Can you please explain why this is correct?

Reply:
> Can you please explain why this is correct?

Could you please clarify? Your comment is on the closing brace.

---
In reply to: [324319040](https://github.com/dotnet/roslyn/pull/38683#discussion_r324319040) [](ancestors = 324319040)

Reply:
I'm referring to the line
`                implementationInInterfacesMightChangeResult = false;`

---
In reply to: [324322301](https://github.com/dotnet/roslyn/pull/38683#discussion_r324322301) [](ancestors = 324322301,324319040)

Reply:
> I'm referring to the line
```implementationInInterfacesMightChangeResult = false;```

We weren't even going to look in interfaces. So, it doesn't matter what is in them.


---
In reply to: [324327053](https://github.com/dotnet/roslyn/pull/38683#discussion_r324327053) [](ancestors = 324327053,324322301,324319040)

Reply:
I don't understand how you conclude "We weren't even going to look in interfaces".  This is the else branch of an if with the condition `(object)implicitImpl == null && seenTypeDeclaringInterface && tryDefaultInterfaceImplementation`

---
In reply to: [324329540](https://github.com/dotnet/roslyn/pull/38683#discussion_r324329540) [](ancestors = 324329540,324327053,324322301,324319040)

Reply:
> I don't understand how you conclude "We weren't even going to look in interfaces".  This is the else branch of an if with the condition (object)implicitImpl == null && seenTypeDeclaringInterface && tryDefaultInterfaceImplementation

As we discussed offline, this code path doesn't depend on value of ```ignoreImplementationInInterfaces``` parameter. The method would produce the same result whether it is true or false. I'll add clarifying comment for the new parameters

---
In reply to: [324332609](https://github.com/dotnet/roslyn/pull/38683#discussion_r324332609) [](ancestors = 324332609,324329540,324327053,324322301,324319040)